definitions/checks/disk/performance.rb

```diff
@@ -0,0 +1,75 @@
+module Checks
+  module Disk
+    class Performance < ForemanMaintain::Check
+      metadata do
+        label :disk_performance
+        description 'Check for recommended disk speed of pulp, mongodb, pgsql dir.'
+        tags :pre_upgrade
+        preparation_steps { Procedures::Packages::Install.new(:packages => %w[hdparm fio]) }
+      end
+
+      EXPECTED_IO = 80
+      DEFAULT_UNIT   = 'MB/sec'.freeze
+      DEFAULT_DIRS   = ['/var/lib/pulp', '/var/lib/mongodb', '/var/lib/pgsql'].freeze
+
+      attr_reader :stats
+
+      def run
+        @stats = {}
+        with_spinner(description) do |spinner|
+          io_obj, success = compute_disk_speed(spinner)
+          spinner.update('Finished')
+          puts "\n"
+          puts show_stats
+
+          assert(success, io_obj.slow_disk_error_msg)
+        end
+      end
+
+      def check_only_single_device?
+        DEFAULT_DIRS.map do |dir|
+          ForemanMaintain::Utils::Disk::Device.new(dir).name
+        end.uniq.length <= 1
+      end
+
+      def dirs_to_check
+        return DEFAULT_DIRS.first(1) if check_only_single_device?
+        DEFAULT_DIRS
+      end
+
+      private
+
+      def compute_disk_speed(spinner)
+        success = true
+        io_obj = ForemanMaintain::Utils::Disk::NilDevice.new
+
+        dirs_to_check.each do |dir|
+          io_obj = ForemanMaintain::Utils::Disk::Device.new(dir)
+
+          spinner.update("[Speed check In-Progress] device: #{io_obj.name}")
+          push_stats(io_obj)
```

Comment:
Currently, I don't see any other place where we have used method `push_stats`.
Instead of creating a method, can we please place code here itself?

Reply:
I feel this to be a good coding practice. For readability.

Reply:
As I write this I think the method name can be improved with `stats <<`